replications/internal/queue_management.go

```diff
@@ -71,44 +81,133 @@ func (qm *durableQueueManager) InitializeQueue(replicationID platform.ID, maxQue
 		return err
 	}
 
-	// Map new durable queue to its corresponding replication stream via replication ID
-	qm.replicationQueues[replicationID] = newQueue
-
 	// Open the new queue
 	if err := newQueue.Open(); err != nil {
 		return err
 	}
 
+	// Map new durable queue and scanner to its corresponding replication stream via replication ID
+	rq := replicationQueue{
+		queue:   newQueue,
+		logger:  qm.logger,
+		done:    make(chan struct{}),
+		receive: make(chan struct{}),
+	}
+	qm.replicationQueues[replicationID] = &rq // todo does this need to be held?
+	rq.Open()
+
 	qm.logger.Debug("Created new durable queue for replication stream",
 		zap.String("id", replicationID.String()), zap.String("path", dir))
 
 	return nil
 }
 
+func (rq *replicationQueue) Open() {
+	rq.wg.Add(1)
+	go rq.run()
+}
+
+func (rq *replicationQueue) Close() error {
+	close(rq.receive)
+	close(rq.done)
+	rq.wg.Wait() // wait for goroutine to finish processing all messages
+	return rq.queue.Close()
+}
+
+func (rq *replicationQueue) run() {
+	defer rq.wg.Done()
+
+	retryInterval := time.Second // todo configurable?
+	retryTimer := time.NewTicker(retryInterval)
+	defer retryTimer.Stop()
+
+	writer := func() {
+		for {
+			_, err := rq.SendWrite(func(b []byte) error {
+				rq.logger.Info("written bytes", zap.String("bytes", string(b)))
+				return nil
+			})
+			if err != nil {
+				if err == io.EOF {
+					// No more data
+					rq.logger.Debug("Finished replication writes for queue")
```

Comment:
Looks like the call to `rq.queue.NewScanner()` returns an EOF if there's no data in the queue, so the debug log gets this line every second, which probably isn't desirable.

Reply:
Yeah, this logging was mostly just for sanity checking that things worked, and should definitely be removed in the future. I can remove it now or we can remove it when the proper calls are hooked in

Reply:
Makes sense - if nothing else a `TODO` would be good so we don't forget to remove it in the future.